### PR TITLE
Travis-CI has not actually been running the tests.

### PR DESCRIPTION
**Pull request by anthonyryan1**
When `./all_tests.py` is run from outside of the tests directory it fails to run the
majority of the test suite. This is either something that needs to be fixed in
all_tests.py or we need to cd into the tests dir before running them.

This pull request does the later, but if someone would rather solve this problem in all_tests.py this can serve as a place to discuss that.

### CODE REVIEW

**Comment by fernandog** [comment]
@anthonyryan1 please check your PR. The Travis CI build failed

**Comment by fernandog** [comment]
```
pi@raspbmc:~/SickRage/tests$ python all_tests.py
E
======================================================================
ERROR: testAll (__main__.AllTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "all_tests.py", line 29, in setUp
    self.suites = [unittest.defaultTestLoader.loadTestsFromName(file_string) for                                                                              file_string in self.module_strings]
  File "/usr/lib/python2.7/unittest/loader.py", line 91, in loadTestsFromName
    module = __import__('.'.join(parts_copy))
  File "/home/pi/SickRage/tests/tv_tests.py", line 21, in <module>
    import test_lib as test
  File "/home/pi/SickRage/tests/test_lib.py", line 33, in <module>
    import sickbeard
  File "/home/pi/SickRage/sickbeard/__init__.py", line 37, in <module>
    from sickbeard import providers, metadata, config, webserveInit
  File "/home/pi/SickRage/sickbeard/providers/__init__.py", line 51, in <module>
    import generic
  File "/home/pi/SickRage/sickbeard/providers/generic.py", line 33, in <module>
    from sickbeard import tvcache
  File "/home/pi/SickRage/sickbeard/tvcache.py", line 35, in <module>
    from sickbeard.rssfeeds import RSSFeeds
  File "/home/pi/SickRage/sickbeard/rssfeeds.py", line 12, in <module>
    from feedcache.cache import Cache
  File "/home/pi/SickRage/lib/feedcache/__init__.py", line 40, in <module>
    from cache import Cache
  File "/home/pi/SickRage/lib/feedcache/cache.py", line 35, in <module>
    from feedparser import feedparser
ImportError: cannot import name feedparser

```

**Comment by anthonyryan1** [comment]
Hello @fernandog,

Believe it or not, it failing is intentional for the purpose of discussing the issue at hand. Take a look at the number of tests being run on both the latest commit and this pull request. The tests are not being run presently so there's actually no indication that the develop HEAD is able to pass it's own tests.

[Current develop HEAD](https://travis-ci.org/SiCKRAGETV/SickRage/jobs/54004163)
-----------------------------
```
==================
STARTING - ALL TESTS
==================

----------------------------------------------------------------------
Ran 0 tests in 0.000s

OK
.

----------------------------------------------------------------------
Ran 1 test in 0.001s

OK
```

[This pull request](https://travis-ci.org/SiCKRAGETV/SickRage/jobs/54019884)
----------------------
```
==================
STARTING - ALL TESTS
==================
- xem_tests.py
- tv_tests.py
- torrent_tests.py
- snatch_tests.py
- scene_helpers_tests.py
- pp_tests.py
- name_parser_tests.py
- issue_submitter_tests.py
- feedparser_tests.py
- encoding_tests.py
- db_tests.py
- config_tests.py
- common_tests.py
.EE.EE.EEEEEE...E.......................
======================================================================
ERROR: test_init_empty_db (tv_tests.TVEpisodeTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/SiCKRAGETV/SickRage/tests/tv_tests.py", line 74, in test_init_empty_db
    ep.saveToDB()
  File "/home/travis/build/SiCKRAGETV/SickRage/sickbeard/tv.py", line 2050, in saveToDB
    myDB.upsert("tv_episodes", newValueDict, controlValueDict)
  File "/home/travis/build/SiCKRAGETV/SickRage/sickbeard/db.py", line 222, in upsert
    self.action(query, valueDict.values() + keyDict.values())
  File "/home/travis/build/SiCKRAGETV/SickRage/sickbeard/db.py", line 170, in action
    sqlResult = self.execute(query, args, fetchall=fetchall, fetchone=fetchone)
  File "/home/travis/build/SiCKRAGETV/SickRage/sickbeard/db.py", line 92, in execute
    raise e
OperationalError: disk I/O error

======================================================================
ERROR: test_change_indexerid (tv_tests.TVShowTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/SiCKRAGETV/SickRage/tests/tv_tests.py", line 47, in test_change_indexerid
    show.saveToDB()
  File "/home/travis/build/SiCKRAGETV/SickRage/sickbeard/tv.py", line 1167, in saveToDB
    myDB.upsert("tv_shows", newValueDict, controlValueDict)
  File "/home/travis/build/SiCKRAGETV/SickRage/sickbeard/db.py", line 222, in upsert
    self.action(query, valueDict.values() + keyDict.values())
  File "/home/travis/build/SiCKRAGETV/SickRage/sickbeard/db.py", line 170, in action
    sqlResult = self.execute(query, args, fetchall=fetchall, fetchone=fetchone)
  File "/home/travis/build/SiCKRAGETV/SickRage/sickbeard/db.py", line 92, in execute
    raise e
OperationalError: disk I/O error

======================================================================
ERROR: test_set_name (tv_tests.TVShowTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/SiCKRAGETV/SickRage/tests/tv_tests.py", line 59, in test_set_name
    show.saveToDB()
  File "/home/travis/build/SiCKRAGETV/SickRage/sickbeard/tv.py", line 1167, in saveToDB
    myDB.upsert("tv_shows", newValueDict, controlValueDict)
  File "/home/travis/build/SiCKRAGETV/SickRage/sickbeard/db.py", line 222, in upsert
    self.action(query, valueDict.values() + keyDict.values())
  File "/home/travis/build/SiCKRAGETV/SickRage/sickbeard/db.py", line 170, in action
    sqlResult = self.execute(query, args, fetchall=fetchall, fetchone=fetchone)
  File "/home/travis/build/SiCKRAGETV/SickRage/sickbeard/db.py", line 92, in execute
    raise e
OperationalError: disk I/O error

======================================================================
ERROR: test_getEpisode (tv_tests.TVTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/SiCKRAGETV/SickRage/tests/tv_tests.py", line 95, in test_getEpisode
    show.saveToDB()
  File "/home/travis/build/SiCKRAGETV/SickRage/sickbeard/tv.py", line 1167, in saveToDB
    myDB.upsert("tv_shows", newValueDict, controlValueDict)
  File "/home/travis/build/SiCKRAGETV/SickRage/sickbeard/db.py", line 222, in upsert
    self.action(query, valueDict.values() + keyDict.values())
  File "/home/travis/build/SiCKRAGETV/SickRage/sickbeard/db.py", line 170, in action
    sqlResult = self.execute(query, args, fetchall=fetchall, fetchone=fetchone)
  File "/home/travis/build/SiCKRAGETV/SickRage/sickbeard/db.py", line 92, in execute
    raise e
OperationalError: disk I/O error

======================================================================
ERROR: test_sceneExceptionByName (scene_helpers_tests.SceneExceptionTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/SiCKRAGETV/SickRage/tests/scene_helpers_tests.py", line 96, in setUp
    scene_exceptions.retrieve_exceptions()
  File "/home/travis/build/SiCKRAGETV/SickRage/sickbeard/scene_exceptions.py", line 182, in retrieve_exceptions
    setLastRefresh(sickbeard.indexerApi(indexer).name)
  File "/home/travis/build/SiCKRAGETV/SickRage/sickbeard/scene_exceptions.py", line 56, in setLastRefresh
    {'list': list})
  File "/home/travis/build/SiCKRAGETV/SickRage/sickbeard/db.py", line 222, in upsert
    self.action(query, valueDict.values() + keyDict.values())
  File "/home/travis/build/SiCKRAGETV/SickRage/sickbeard/db.py", line 170, in action
    sqlResult = self.execute(query, args, fetchall=fetchall, fetchone=fetchone)
  File "/home/travis/build/SiCKRAGETV/SickRage/sickbeard/db.py", line 92, in execute
    raise e
OperationalError: disk I/O error

======================================================================
ERROR: test_sceneExceptionByNameEmpty (scene_helpers_tests.SceneExceptionTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/SiCKRAGETV/SickRage/tests/scene_helpers_tests.py", line 96, in setUp
    scene_exceptions.retrieve_exceptions()
  File "/home/travis/build/SiCKRAGETV/SickRage/sickbeard/scene_exceptions.py", line 182, in retrieve_exceptions
    setLastRefresh(sickbeard.indexerApi(indexer).name)
  File "/home/travis/build/SiCKRAGETV/SickRage/sickbeard/scene_exceptions.py", line 56, in setLastRefresh
    {'list': list})
  File "/home/travis/build/SiCKRAGETV/SickRage/sickbeard/db.py", line 222, in upsert
    self.action(query, valueDict.values() + keyDict.values())
  File "/home/travis/build/SiCKRAGETV/SickRage/sickbeard/db.py", line 170, in action
    sqlResult = self.execute(query, args, fetchall=fetchall, fetchone=fetchone)
  File "/home/travis/build/SiCKRAGETV/SickRage/sickbeard/db.py", line 92, in execute
    raise e
OperationalError: disk I/O error

======================================================================
ERROR: test_sceneExceptionsBabylon5 (scene_helpers_tests.SceneExceptionTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/SiCKRAGETV/SickRage/tests/scene_helpers_tests.py", line 96, in setUp
    scene_exceptions.retrieve_exceptions()
  File "/home/travis/build/SiCKRAGETV/SickRage/sickbeard/scene_exceptions.py", line 182, in retrieve_exceptions
    setLastRefresh(sickbeard.indexerApi(indexer).name)
  File "/home/travis/build/SiCKRAGETV/SickRage/sickbeard/scene_exceptions.py", line 56, in setLastRefresh
    {'list': list})
  File "/home/travis/build/SiCKRAGETV/SickRage/sickbeard/db.py", line 222, in upsert
    self.action(query, valueDict.values() + keyDict.values())
  File "/home/travis/build/SiCKRAGETV/SickRage/sickbeard/db.py", line 170, in action
    sqlResult = self.execute(query, args, fetchall=fetchall, fetchone=fetchone)
  File "/home/travis/build/SiCKRAGETV/SickRage/sickbeard/db.py", line 92, in execute
    raise e
OperationalError: disk I/O error

======================================================================
ERROR: test_sceneExceptionsEmpty (scene_helpers_tests.SceneExceptionTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/SiCKRAGETV/SickRage/tests/scene_helpers_tests.py", line 96, in setUp
    scene_exceptions.retrieve_exceptions()
  File "/home/travis/build/SiCKRAGETV/SickRage/sickbeard/scene_exceptions.py", line 182, in retrieve_exceptions
    setLastRefresh(sickbeard.indexerApi(indexer).name)
  File "/home/travis/build/SiCKRAGETV/SickRage/sickbeard/scene_exceptions.py", line 56, in setLastRefresh
    {'list': list})
  File "/home/travis/build/SiCKRAGETV/SickRage/sickbeard/db.py", line 222, in upsert
    self.action(query, valueDict.values() + keyDict.values())
  File "/home/travis/build/SiCKRAGETV/SickRage/sickbeard/db.py", line 170, in action
    sqlResult = self.execute(query, args, fetchall=fetchall, fetchone=fetchone)
  File "/home/travis/build/SiCKRAGETV/SickRage/sickbeard/db.py", line 92, in execute
    raise e
OperationalError: disk I/O error

======================================================================
ERROR: test_sceneExceptionsResetNameCache (scene_helpers_tests.SceneExceptionTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/SiCKRAGETV/SickRage/tests/scene_helpers_tests.py", line 96, in setUp
    scene_exceptions.retrieve_exceptions()
  File "/home/travis/build/SiCKRAGETV/SickRage/sickbeard/scene_exceptions.py", line 182, in retrieve_exceptions
    setLastRefresh(sickbeard.indexerApi(indexer).name)
  File "/home/travis/build/SiCKRAGETV/SickRage/sickbeard/scene_exceptions.py", line 56, in setLastRefresh
    {'list': list})
  File "/home/travis/build/SiCKRAGETV/SickRage/sickbeard/db.py", line 222, in upsert
    self.action(query, valueDict.values() + keyDict.values())
  File "/home/travis/build/SiCKRAGETV/SickRage/sickbeard/db.py", line 170, in action
    sqlResult = self.execute(query, args, fetchall=fetchall, fetchone=fetchone)
  File "/home/travis/build/SiCKRAGETV/SickRage/sickbeard/db.py", line 92, in execute
    raise e
OperationalError: disk I/O error

======================================================================
ERROR: test_allPossibleShowNames (scene_helpers_tests.SceneTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/SiCKRAGETV/SickRage/tests/scene_helpers_tests.py", line 74, in test_allPossibleShowNames
    myDB.action("INSERT INTO scene_exceptions (indexer_id, show_name, season) VALUES (?,?,?)", [-1, 'Exception Test', -1])
  File "/home/travis/build/SiCKRAGETV/SickRage/sickbeard/db.py", line 170, in action
    sqlResult = self.execute(query, args, fetchall=fetchall, fetchone=fetchone)
  File "/home/travis/build/SiCKRAGETV/SickRage/sickbeard/db.py", line 92, in execute
    raise e
OperationalError: disk I/O error

======================================================================
ERROR: test_process (pp_tests.PPBasicTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/SiCKRAGETV/SickRage/tests/pp_tests.py", line 50, in test_process
    show.saveToDB()
  File "/home/travis/build/SiCKRAGETV/SickRage/sickbeard/tv.py", line 1167, in saveToDB
    myDB.upsert("tv_shows", newValueDict, controlValueDict)
  File "/home/travis/build/SiCKRAGETV/SickRage/sickbeard/db.py", line 222, in upsert
    self.action(query, valueDict.values() + keyDict.values())
  File "/home/travis/build/SiCKRAGETV/SickRage/sickbeard/db.py", line 170, in action
    sqlResult = self.execute(query, args, fetchall=fetchall, fetchone=fetchone)
  File "/home/travis/build/SiCKRAGETV/SickRage/sickbeard/db.py", line 92, in execute
    raise e
OperationalError: disk I/O error

----------------------------------------------------------------------
Ran 40 tests in 2.133s

FAILED (errors=11)
E
======================================================================
ERROR: testAll (__main__.AllTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "./all_tests.py", line 41, in testAll
    sys.exit(-1)
SystemExit: -1

----------------------------------------------------------------------
Ran 1 test in 4.795s

FAILED (errors=1)
```

**Comment by fernandog** [comment]
I get that bunch of errors too.
its strange you dont jave import error. Maybe running  newer/older version of the lib?

do we need to run this tests in any specific occasion? Like every boot or just when needed?
Dont know about it

**Comment by TagForce** [comment]
@fernandog Travis runs the tests in the tests folder to ensure that a PR actually doesn't break some fundamental functions in SR (the functions for which a test script was created), and it will only pass when those tests are succesful. If they're not, it warns the developers of possible issues with that particular test. Since the tests have never been actually run when submitting PRs, it would not encounter errors, so if it could actually build the install with the info from the PR it would pass.

Seeing as these tests have not been updated in a long time, it might be a smart idea to go through them and make sure they are still current.

Once this is done, we should also create new tests for major new features, so they get testes as well.

Basically, it just saves us from debugging PRs before merging, as travis makes sure the basic functions in SR aren't broken.

**Comment by fernandog** [comment]
the issue submitter works. It created a fake issue. IMO it shouldnt run always! Or create a closed issue.
i wont look a this now. Only when all known issues are closed

**Comment by TagForce** [comment]
@fernandog What are you talking about?
The tests are only run when you either start them manually (from commandline), or when you submit a PR during the Travis CI build. Has nothing to do with the issue submitter :)

**Comment by fernandog** [comment]
@TagForce one of the tests, is the issue submitter test. it create a FAKE EXCEPTION to test the creation of the issue:
https://github.com/SiCKRAGETV/sickrage-issues/issues?q=app+%22fake+exception%22+

**Comment by TagForce** [comment]
@fernandog Ahh... I missed that one

**Comment by anthonyryan1** [comment]
If you want to avoid running tests after each commit and pull request, why don't you disable Travis CI? It's better than this current process of not running the tests at all while marking every commit and build as passing them.

**Comment by anthonyryan1** [comment]
I have updated this pull request with another possible solution to the particular blocker you mentioned. I have added a blacklist to all_tests.py (and taken the liberty of making it +x by default) so that the `issue_submitter_tests.py` group can be skipped while the remaining tests can all be run.

**Comment by miigotu** [comment]
I've been poking at this today. All failures are related to db calls once you set the imports correctly. I am assuming a missed connection.close() or db is not connected/initiated yet, as it rarely creates the cache.db/failed.db
